### PR TITLE
Small fixes mixed in one pull request

### PR DESCRIPTION
**Pull request by mchindri**
Each fix description if found in its own commit.

### CODE REVIEW

File: include/error.h [hunk 1]
```diff
@@ -51,4 +51,7 @@
 #define FAILURE		-1
 #endif
 
+#define RET_OK(x)	((x) == SUCCESS)
+#define RET_FAIL(x)	((x) != SUCCESS)
```
Comment (adrimbarean) [inline include/error.h]: Does it make it more readable? I feel like more often than not abstracting things can create confusion. That said I do not oppose the change, I would just wish to know if the majority of us see improvement here.

Comment (mchindri) [reply include/error.h]: I will put this in other pull request

Comment (cristiantpop) [reply include/error.h]: I wouldn't define this macros. They might create confusion.

**Comment by mchindri** [comment]
V2
- Remove error.h commit and move to https://github.com/analogdevicesinc/no-OS/pull/509

**Pull request merged**